### PR TITLE
Docker compose

### PR DESCRIPTION
**Pull request by N-E-W-T-O-N**
I have created a docker-compose.yml for a better utilization of Dockerfile.

@dualvtable please let me know for any improvement.

### CODE REVIEW

**Review by jackos — CHANGES_REQUESTED** [review]
Hi this is great thanks @N-E-W-T-O-N builds and starts the container in a single command. The files have moved around since you raised this though. Do you mind placing the docker-compose.yaml into `examples/docker` please?

**Comment by N-E-W-T-O-N** [comment]
Ok I did the little rearangement as you asked

File: examples/docker/build-image.sh [hunk 1]
```diff
@@ -1,65 +1,65 @@
-#!/bin/sh
```
Comment (jackos) [inline examples/docker/build-image.sh:1]: All the line endings in this file have changed to windows style `CR` `LF`, which makes it fail on Linux. You can revert this from the base of the repo by doing `git checkout main -- examples/docker/build-image.sh`

**Review by jackos — CHANGES_REQUESTED** [review]
Everything else looks good, thanks again.

File: examples/docker/build-image.sh (hunk 1 above)
Comment (N-E-W-T-O-N) [reply examples/docker/build-image.sh]: Change the end of line sequence as asked.

**Pull request merged**